packages/db-commons/src/getEnv.cjs

```diff
@@ -0,0 +1,32 @@
+/**
+ * @param {string[]} key
+ * @param {any} envMap
+ */
+const getEnv = (envMap, ...key) => {
+	const keyArray = [...key];
+
+	let t = envMap;
+	while (keyArray.length) {
+		t = t[keyArray.shift()];
+		if (Array.isArray(t) && keyArray.length) {
+			// We're too soon
+			throw new Error(`Could not find ${key} in env map!`);
+		}
+	}
+	if (Array.isArray(t)) {
+		// We have run through the entire key and landed on an array
+		for (const k of t) {
+			if (process.env[k.key]) {
+				if (k.deprecated)
+					console.warn(
+						`Found environment variable ${k.key}. It is being used but is currently deprecated.`
+					);
+				return process.env[k.key];
+			}
+		}
+	} else {
+		throw new Error(`Could not find ${key} in env map!`);
+	}
+};
+
+exports.getEnv = getEnv;
```

Comment:
Looking at the tests, it seems to me that the `key` parameter is suppose to represent a hierarchy.  e.g for the following `envMap`, if we want `2nd`, we ought to call `getEnv(envMap, ['1st', '2nd'])`.  I think some of the vars/params in getEnv can be renamed to make this a bit more self documenting.

		const map = {
			'1st': {
				'2nd': [{ key: 'my-key', deprecated: false }]
			}
		};

Reply:
In one or two places nesting was used; down for that rename though